test/hexer_docs_SUITE.erl

```diff
@@ -0,0 +1,189 @@
+  meck:expect(hexer_utils, cmd, OsFalseFun),
+  ok = try ok = hexer_docs:publish(), error
+       catch _:{hexer_docs, {bad_github_tag, ErrorTag}} -> ok
+       end,
+  {comment, ""}.
+
+-spec not_published(hexer_test_utils:config()) -> {comment, string()}.
+not_published(_Config) ->
+  ok = run_in_dir(app_directory(), fun generate_api_key/0),
+  PromptMockFalseFun = fun(_, _) -> false end,
+  meck:expect(hexer_utils, prompt, PromptMockFalseFun),
+  mock_shotgun_ok(),
+  ct:comment("Don't proceed publishing"),
+  ok = run_in_dir(app_directory(), fun hexer_docs:publish/0),
+  timeout = hexer_test_utils:wait_receive(publish, 500),
+  {comment, ""}.
+
+-spec no_files_to_upload(hexer_test_utils:config()) -> {comment, string()}.
+no_files_to_upload(_Config) ->
+  ct:comment("Error: app_file_not_found error"),
+   ok = generate_api_key(),
+  run_in_dir(app_directory(), fun delete_docs/0),
+  PromptMockTrueFun = fun(_, _) -> true end,
+  meck:expect(hexer_utils, prompt, PromptMockTrueFun),
+  mock_shotgun_ok(),
+  ok = try ok = hexer_docs:publish(), error
+       catch _:{hexer_docs, no_files_to_upload} -> ok
+       end,
+  {comment, ""}.
+
+-spec shotgun_error(hexer_test_utils:config()) -> {comment, string()}.
+shotgun_error(_Config) ->
+  ct:comment("Error: shotgun error"),
+  ok = run_in_dir(app_directory(), fun generate_api_key/0),
+  run_in_dir(app_directory(), fun create_docs/0),
+  PromptMockTrueFun = fun(_, _) -> true end,
+  meck:expect(hexer_utils, prompt, PromptMockTrueFun),
+  PostErrorFun = fun(_, _, _, _, _) -> {error, unexpected} end,
+  meck:expect(shotgun, post, PostErrorFun),
+  ok = try ok = run_in_dir(app_directory(), fun hexer_docs:publish/0), error
+       catch _:unexpected -> ok
+       end,
+  {comment, ""}.
+
+-spec server_error(hexer_test_utils:config()) -> {comment, string()}.
+server_error(_Config) ->
+  ok = run_in_dir(app_directory(), fun generate_api_key/0),
+  run_in_dir(app_directory(), fun create_docs/0),
+  PromptMockTrueFun = fun(_, _) -> true end,
+  meck:expect(hexer_utils, prompt, PromptMockTrueFun),
+  ct:comment("Error: server error when publishing"),
+  ResponseError = #{status_code => 500, body => term_to_binary({})},
+  Post500Fun = fun(_, _, _, _, _) -> {ok, ResponseError} end,
+  meck:expect(shotgun, post, Post500Fun),
+  ok = try ok = run_in_dir(app_directory(), fun hexer_docs:publish/0), error
+       catch _:{500, {}} -> ok
+       end,
+  {comment, ""}.
+
+-spec publish_succeeds(hexer_test_utils:config()) -> {comment, string()}.
+publish_succeeds(_Config) ->
+  ok = run_in_dir(app_directory(), fun generate_api_key/0),
+  PromptMockTrueFun = fun(_, _) -> true end,
+  meck:expect(hexer_utils, prompt, PromptMockTrueFun),
+  mock_shotgun_ok(),
+  ct:comment("Publishing succeeds"),
+  ok = run_in_dir(app_directory(), fun hexer_docs:publish/0),
+  ok = hexer_test_utils:wait_receive(publish, 500),
+
+  {comment, ""}.
+
+-spec run_in_dir(string(), function()) -> any().
```

Comment:
Most of the functions from here on in this file are the same as in the `hexer_package_SUITE`, it's better if we move them to a common `hexer_test_utils` module.